src/main/scanLogic/scanRunners/eosScan.ts

```diff
@@ -46,44 +47,22 @@ export interface EosIssueLocation {
 }
 
 export class EosRunner extends BinaryRunner {
-    private static readonly BINARY_FOLDER: string = 'eos';
-    private static readonly BINARY_NAME: string = 'eos_scanner';
-
-    constructor(connectionManager: ConnectionManager, abortCheckInterval: number, logManager: LogManager) {
-        super(
-            connectionManager,
-            abortCheckInterval,
-            ScanType.Eos,
-            logManager,
-            new Resource('', path.join(ScanUtils.getHomePath(), EosRunner.BINARY_FOLDER, EosRunner.getBinaryName()), logManager)
-        );
+    constructor(
+        connectionManager: ConnectionManager,
+        logManager: LogManager,
+        binary?: Resource,
+        timeout: number = ScanUtils.ANALYZER_TIMEOUT_MILLISECS
+    ) {
+        super(connectionManager, timeout, ScanType.Eos, logManager, binary);
     }
 
-    public validateSupported(): boolean {
-        if (os.platform() !== 'linux' && os.platform() !== 'darwin' && os.platform() !== 'win32') {
-            this._logManager.logMessage("Eos scan is not supported on '" + os.platform() + "' os", 'DEBUG');
-            return false;
-        }
-        return super.validateSupported();
-    }
-
-    /** @override */
-    protected static getBinaryName(): string {
-        let name: string = EosRunner.BINARY_NAME;
-        switch (os.platform()) {
-            case 'linux':
-                return name + '_ubuntu';
-            case 'darwin':
-                return name + '_macos';
-            case 'win32':
-                return name + '.exe';
-        }
-        return name;
+    public static supportedLanguages(): LanguageType[] {
+        return ['python'];
```

Comment:
Don't we have an enum with all languages?

Reply:
we have packgeType, those are different from the actual languages that the files written in (some of them are the same like python, but some are different like maven, yarn....) so I added a new type 'LanguageType' for this